Build failure when applied after the 'eclipse' plugin

I've just noticed that mixingradle 0.7-SNAPSHOT causes a build failure if the '**eclipse**' plugin is applied **before** the '**org.spongeforge.mixin**' plugin (it works fine the other way around).

It would be nice to see this fixed, as finding the cause was really time consuming with such an obscure error message from gradle.

Error message:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':forge-1.19:eclipseFactoryPath' (type 'MixinEclipse.EclipseFactoryPath').
  - In plugin 'org.spongepowered.mixin' type 'org.spongepowered.asm.gradle.plugins.MixinEclipse.EclipseFactoryPath' property 'config' has @Input annotation used on property of type 'Configuration'.

    Reason: A property of type 'Configuration' annotated with @Input cannot determine how to interpret the file.

    Possible solutions:
      1. Annotate with @InputFile for regular files.
      2. Annotate with @InputDirectory for directories.
      3. If you want to track the path, return File.absolutePath as a String and keep @Input.

    Please refer to https://docs.gradle.org/7.5/userguide/validation_problems.html#incorrect_use_of_input_annotation for more details about this problem.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5/userguide/command_line_interface.html#sec:command_line_warnings
```

--- a/README.md
+++ b/README.md
@@ -42,6 +42,12 @@ Please ensure you are using the correct version of MixinGradle for your ForgeGra
  apply plugin: 'org.spongepowered.mixin'
  ```
  
+If using Eclipse, you should also enable the Eclipse APT plugin to receive annotation processor support within Eclipse:  
+  
+ ```groovy
+ apply plugin: 'com.diffplug.eclipse.apt'
+ ```
+  
 3. Create your `mixin` block, specify which sourceSets to process and provide refmap resource names for each one, the generated refmap will be added to the compiler task outputs automatically.
  
  ```groovy

--- a/src/main/groovy/org/spongepowered/asm/gradle/plugins/MixinEclipse.groovy
+++ b/src/main/groovy/org/spongepowered/asm/gradle/plugins/MixinEclipse.groovy
@@ -52,21 +52,28 @@ public class MixinEclipse {
             project.logger.lifecycle '[MixinGradle] Skipping eclipse integration, extension not found'
             return
         }
-        
-        eclipseModel.jdt.file.withProperties { it.setProperty('org.eclipse.jdt.core.compiler.processAnnotations', 'enabled') }
-        def settings = project.tasks.register('eclipseJdtApt', EclipseJdtAptTask.class) {
+
+        def eclipseAptPlugin = project.plugins.findPlugin('com.diffplug.eclipse.apt')
+
+        def settings = project.tasks.register('mixinEclipseJdtApt', EclipseJdtAptTask.class) {
+            inputs.files project.tasks.createSrgToMcp
+            inputs.files project.tasks.createMcpToSrg
             description = 'Creates the Eclipse JDT APT settings file'
             output = project.file('.settings/org.eclipse.jdt.apt.core.prefs')
             mappingsIn = extension.mappings
+            hasDiffplug = eclipseAptPlugin
         }
-        project.tasks.eclipse.dependsOn(settings)
-        
-        def factories = project.tasks.register('eclipseFactoryPath', EclipseFactoryPath.class) {
+
+        def factories = project.tasks.register('mixinEclipseFactorypath', EclipseFactoryPath.class) {
             config = project.configurations.annotationProcessor
             output = project.file('.factorypath')
         }
-        project.tasks.eclipse.dependsOn(factories)
-        
+
+        // These tasks don't do any useful work unless the diffplug APT plugin is applied.
+        if (eclipseAptPlugin) {
+            project.tasks.eclipseJdtApt.dependsOn(settings)
+            project.tasks.eclipseFactorypath.dependsOn(factories)
+        }
     }
     
     static class OrderedProperties extends Properties {
@@ -90,25 +97,28 @@ public class MixinEclipse {
     
     static class EclipseJdtAptTask extends DefaultTask {
         @InputFile File mappingsIn
-        @InputFile File refmapOut = project.file("build/${name}/mixins.refmap.json")
-        @InputFile File mappingsOut = project.file("build/${name}/mixins.mappings.tsrg")
+        @OutputFile File refmapOut = project.file("build/${name}/mixins.refmap.json")
+        @OutputFile File mappingsOut = project.file("build/${name}/mixins.mappings.tsrg")
         @Input Map<String, String> processorOptions = new TreeMap<>()
         
-        @InputFile File genTestDir = project.file('build/.apt_generated_test')
-        @InputFile File genDir = project.file('build/.apt_generated')
+        @OutputFile File genTestDir = project.file('build/.apt_generated_test')
+        @OutputFile File genDir = project.file('build/.apt_generated')
         
         @OutputFile File output
         
+        @Input boolean hasDiffplug = false;
         
         @TaskAction
         def run() {
             MixinExtension extension = project.extensions.findByType(MixinExtension.class)
             def props = new OrderedProperties()
-            props.put('eclipse.preferences.version', '1')
-            props.put('org.eclipse.jdt.apt.aptEnabled', 'true')
-            props.put('org.eclipse.jdt.apt.reconcileEnabled', 'true')
-            props.put('org.eclipse.jdt.apt.genSrcDir', genDir.canonicalPath)
-            props.put('org.eclipse.jdt.apt.genSrcTestDir', genTestDir.canonicalPath)
+            if (!hasDiffplug) {
+                props.put('eclipse.preferences.version', '1')
+                props.put('org.eclipse.jdt.apt.aptEnabled', 'true')
+                props.put('org.eclipse.jdt.apt.reconcileEnabled', 'true')
+                props.put('org.eclipse.jdt.apt.genSrcDir', genDir.canonicalPath)
+                props.put('org.eclipse.jdt.apt.genSrcTestDir', genTestDir.canonicalPath)
+            }
             props.arg('reobfTsrgFile', mappingsIn.canonicalPath)
             props.arg('outTsrgFile', mappingsOut.canonicalPath)
             props.arg('outRefMapFile', refmapOut.canonicalPath)
@@ -153,7 +163,7 @@ public class MixinEclipse {
             }
             */
             
-            props.store(output.newWriter(), null)
+            props.store(output.newWriter(hasDiffplug), null);
         }
     }
 

